### PR TITLE
Add missing move constructor to S2Polyline

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

File: src/s2/s2polyline.h [hunk 1]
```diff
@@ -46,6 +46,7 @@ class S2Polyline final : public S2Region {
   // Creates an empty S2Polyline that should be initialized by calling Init()
   // or Decode().
   S2Polyline();
+  S2Polyline(S2Polyline&&);
```
Comment (jmr) [inline src/s2/s2polyline.h]: Do we also need to add a move `operator=`?

File: src/s2/s2polyline.cc [hunk 2]
```diff
@@ -50,6 +50,13 @@ static const unsigned char kCurrentLosslessEncodingVersionNumber = 1;
 S2Polyline::S2Polyline()
   : s2debug_override_(S2Debug::ALLOW) {}
 
+S2Polyline::S2Polyline(S2Polyline&& other)
+ : s2debug_override_(other.s2debug_override_),
+   num_vertices_(other.num_vertices_),
+   vertices_(std::move(other.vertices_)) {
+  other.num_vertices_ = 0;
```
Comment (jmr) [inline src/s2/s2polyline.cc]: It might be safer/more obvious to do `other_.vertices_.clear()` to get back to the normal empty state.

Comment (danielhlarkin) [reply src/s2/s2polyline.cc]: I don't think it's necessary to add a `release()` operation here. It's explicit in the standard that moving a `unique_ptr` has the same effect as calling `release()` on the source operand.

File: src/s2/s2polyline.h [hunk 3]
```diff
@@ -47,6 +47,10 @@ class S2Polyline final : public S2Region {
   // or Decode().
   S2Polyline();
 
+// S2Polyline is movable, but only privately copyable.
```
Comment (jmr) [inline src/s2/s2polyline.h]: Sorry, I wasn't clear -- with normal two-space indentation.

Comment (danielhlarkin) [reply src/s2/s2polyline.h]: Ah, crap. My bad, the lack of indentation was unintentional.

File: src/s2/s2polyline.h [hunk 4]
```diff
@@ -47,6 +47,10 @@ class S2Polyline final : public S2Region {
   // or Decode().
   S2Polyline();
 
+  // S2Polyline is movable, but only privately copyable.
+  S2Polyline(S2Polyline&&);
+  S2Polyline& operator=(S2PolyLine&&);
```
Comment (jmr) [inline src/s2/s2polyline.h]: s/Line/line/

**Pull request merged**